File: tests/fixtures/resend/resend-marketing-missing-unsubscribe-fixed/02-transactional-no-unsub.ts
import { Resend } from 'resend';

const resend = new Resend(process.env.RESEND_API_KEY);

// Adversarial: a transactional welcome email with no unsubscribe mechanism.
// Transactional mail is not marketing, so the absence is correct and must not
// be flagged.
export async function sendWelcome(email: string) {
  return resend.emails.send({
    from: 'Acme <onboarding@acme.com>',
    to: [email],
    subject: 'Welcome to Acme',
    html: '<p>Welcome aboard!</p>',
    tags: [{ name: 'category', value: 'welcome' }],
  });
}

File: tests/fixtures/resend/resend-marketing-missing-unsubscribe-fixed/02-transactional-no-unsub.py
import os
import resend

resend.api_key = os.environ["RESEND_API_KEY"]

# Adversarial: transactional welcome email (and HTML that mentions "newsletter")
# with no unsubscribe. Must not flag — only marketing tags trigger the rule.
resend.Emails.send(
    {
        "from": "Acme <onboarding@acme.com>",
        "to": ["user@example.com"],
        "subject": "Welcome to Acme",
        "html": "<p>Welcome aboard! Confirm your newsletter waitlist separately.</p>",
        "tags": [{"name": "category", "value": "welcome"}],
    }
)
